tests/shell/shell_recorder/README.md

```diff
@@ -0,0 +1,285 @@
+# Shell Recorder #
+
+## Configuration ##
+
+### Mountpoint: ###
+
+This is the only configuration Variable that has to be set. It is used to determine where the shell_recorder should look for changes.
+e.g. `Mountoint: user/test` tells the shell_recorder that you will be working under `user/test`.
+
+### DiffType: ###
+
+Available: 
+
+`Dump`, `Ini` exports the keys below `Mountpoint` using either the dump or ini format and diffs the changes.
+`File` does a diff on the configuration file mounted to `Mountpoint`.
+
+### File: ###
+
+Tells the shell_recorder what file to use for diffs.
+
+
+
+## Checks ##
+
+Posix-extended regex is used to check and validate return values and outputs.
+
+Options: 
+
+* STDOUT:
+* STDERR:
+* RET:
+* WARNINGS:
+* ERRORS:
+* DIFF:
+
+## Commands ##
+
+A command starts with `<` followed by the usual kdb commands. 
+
+
+## Examples ##
+
+### Basic ###
+
+```
+ % cat db_changes.dat
+
+DiffType: Dump
+Mountpoint: user
+
+DIFF: (> keyNew.*user/test/key)
+< set /test/key val
+
+DIFF: (> keyNew.*user/test/key3)
+< set /test/key2 val2
+
+ % ./shell_recorder.sh db_changes.dat
```

Comment:
Some explanation would be useful would you wanted to achieve here. Its unclear that a file "protocol" is recorded here.